release/nightly_tests/dataset/aggregate_benchmark.py

```diff
@@ -0,0 +1,123 @@
+from typing import Tuple
+
+import ray
+from ray.data.aggregate import _AggregateOnKeyBase, Max, Mean, Min, Sum
+from ray.data.block import Block, KeyFn
+from ray.data.dataset import Dataset
+import pyarrow.compute as pac
+
+from benchmark import Benchmark
+
+
+def run_h2oai(benchmark: Benchmark):
+    """This benchmark is originally from https://github.com/h2oai/db-benchmark
+
+    Here we run all group-by queries from the benchmark on Ray Datasets.
+    The input files are pre-generated and stored in AWS S3 beforehand.
+    """
+    test_input = [
+        ("s3://air-example-data/h2oai_benchmark/G1_1e7_1e2_0_0.csv", "h2oai-500M")
```

Comment:
Is the input really CSV instead of parquet? That seems like it will spend a lot of time decoding just the CSV.

Reply:
@ericl - yes the input is CSV file - [script to generate input file](https://github.com/h2oai/db-benchmark/blob/master/_data/groupby-datagen.R#L49), and [Spark script to run the benchmark](https://github.com/h2oai/db-benchmark/blob/master/spark/groupby-spark.py#L39).

> That seems like it will spend a lot of time decoding just the CSV.

That's true, it will be significantly slower than Parquet. But it's just loaded once, and reused across benchmark runs. And the read time is not accounted into benchmark runtime, same to how h2oai db-benchmark measures other system (e.g. Spark script above). Right now read 500MB takes less than 10 seconds, and 5GB takes less than 1 minute.